packages/blockchain-extension/extension/explorer/environmentExplorer.ts

```diff
@@ -209,71 +214,139 @@ export class BlockchainEnvironmentExplorerProvider implements BlockchainExplorer
-
         const tree: BlockchainTreeItem[] = [];
 
-        try {
-            const environmentEntries: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll();
-
-            if (environmentEntries.length === 0) {
-                const command: vscode.Command = {
-                    command: ExtensionCommands.ADD_ENVIRONMENT,
-                    title: '',
-                    arguments: []
-                };
-                tree.push(new TextTreeItem(this, 'Click + to add environments', command));
-            } else {
-                for (const environmentEntry of environmentEntries) {
-                    if (environmentEntry.managedRuntime) {
+        const environmentEntries: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll();
+
+        const environmentGroups: Array<FabricEnvironmentRegistryEntry[]> = [];
+        const ibmCloudEnvironments: Array<FabricEnvironmentRegistryEntry> = [];
+        const otherEnvironments: Array<FabricEnvironmentRegistryEntry> = [];
+        for (const environment of environmentEntries) {
+            if (environmentGroups.length === 0) {
+                if (environment.environmentType === EnvironmentType.LOCAL_ENVIRONMENT) {
+                    environmentGroups.push([environment]);
+                } else if (environment.environmentType === EnvironmentType.OPS_TOOLS_ENVIRONMENT || environment.environmentType === EnvironmentType.SAAS_OPS_TOOLS_ENVIRONMENT) {
+                    ibmCloudEnvironments.push(environment);
+                } else {
+                    otherEnvironments.push(environment);
+                }
+                continue;
+            }
 
-                        let runtime: LocalEnvironment | ManagedAnsibleEnvironment;
-                        if (environmentEntry.environmentType === EnvironmentType.LOCAL_ENVIRONMENT) {
-                            runtime = await LocalEnvironmentManager.instance().ensureRuntime(environmentEntry.name, undefined, environmentEntry.numberOfOrgs);
-                        } else {
-                            // Managed ansible
-                            runtime = ManagedAnsibleEnvironmentManager.instance().ensureRuntime(environmentEntry.name, environmentEntry.environmentDirectory);
-                        }
+            // Used to check if group exists already
+            const groupIndex: number = environmentGroups.findIndex((group: FabricEnvironmentRegistryEntry[]) => {
+                return group[0].environmentType === environment.environmentType;
+            });
+
+            if (groupIndex !== -1) {
+                // If a group with the same environmentType exists, then push gateway to the group
+                environmentGroups[groupIndex].push(environment);
+            } else { // if there's a local env that group will have been created first, so check for other env types
```

Comment:
If I don't have a 1 Org Local Fabric (or any other local environments), then surely this will break when I decide to add a local environment?

Reply:
Oh I see what you mean by your comment - disregard this!